meta-mender-core/conf/mender-vars.json

```diff
@@ -0,0 +1,83 @@
+{
+    "__MENDER__INFO__": 
+    ["This file contains all mender variable definitions", 
+     "used by meta-mender layer, and is used the python-", 
+     "event handler function in classes/mender-setup.bbclass", 
+     "that run after bitbake is finished parsing", 
+     "This function compares all MENDER_.* variables in", 
+     "current build and check if it matches any of the", 
+     "keys below. If a variable is found but not listed", 
+     "here; bitbake issues a warning.", 
+     "If a value is specified, it is compared using", 
+     "regular expression syntax with the value provided", 
+     "in current build; furthermore if a value is a list", 
+     "of regular expressions, each value is checked for", 
+     "existance in the variable in current build", 
+     "For example: we can issue a warning if user does",
+     "not include mender-uboot and mender-install",
+     "if we give the following definition below:", 
+     "'MENDER_FEATURES_ENABLE': ['mender-uboot', 'mender-install']",
+     "(using single quote (') for double quote in the above string)"],
+    "MENDER_ARTIFACT_NAME": "",
+    "MENDER_BOOT_PART": "",
+    "MENDER_BOOT_PART_DEFAULT": "",
+    "MENDER_BOOT_PART_DEFAULT_mender-ubi": "",
```

Comment:
We will need to handle this specially: String like `_mender-ubi` are called an override, and there is an infinite number of possibilities for such strings, because the string can refer to some property of the build. This one in particular is "if we are building mender with ubi", but it can be anything, for example `_64bit`. The result then ends up in `MENDER_BOOT_PART_DEFAULT` if the property is true.

So I think we should remove all the `mender-ubi` strings, and before doing the matching, simply drop everything from the variable from the point where is an underscore followed by a lowercase letter.